src/formatting/expr.rs

```diff
@@ -127,9 +127,18 @@ pub(crate) fn format_expr(
         ast::ExprKind::Block(ref block, opt_label) => {
             match expr_type {
                 ExprType::Statement => {
-                    if let rw @ Some(_) =
-                        rewrite_empty_block(context, block, Some(&expr.attrs), opt_label, "", shape)
-                    {
+                    let empty_block_prefix = match block.rules {
+                        ast::BlockCheckMode::Unsafe(..) => "unsafe ",
+                        _ => "",
+                    };
```

Comment:
Note that the `is_unsafe_block` helper function exists and should be used instead

Reply:
Will do thanks